.circleci/config.yml

```diff
@@ -462,6 +478,20 @@ jobs:
       - store_artifacts: *artifacts_solc
       - persist_to_workspace: *artifacts_executables
 
+  b_ubu_ubsan_clang: &build_ubuntu2004_clang
+    docker:
+      - image: << pipeline.parameters.ubuntu-2004-clang-docker-image >>
+    environment:
+      CC: clang
+      CXX: clang++
+      CMAKE_OPTIONS: -DSANITIZE=undefined
+      MAKEFLAGS: -j 3
+    steps:
+      - checkout
+      - run: *run_build
+      - store_artifacts: *artifacts_solc
+      - persist_to_workspace: *artifacts_executables
```

Comment:
You could parametrize it and merge it with `b_ubu_asan_clang`:

```yaml
  b_ubu_san_clang: &build_ubuntu2004_clang
    parameters:
      sanitizer:
        type: string
    docker:
      - image: << pipeline.parameters.ubuntu-2004-clang-docker-image >>
    environment:
      CC: clang
      CXX: clang++
      CMAKE_OPTIONS: -DSANITIZE=<< parameters.sanitizer >>
      MAKEFLAGS: -j 3
    steps:
      - checkout
      - run: *run_build
      - store_artifacts: *artifacts_solc
      - persist_to_workspace: *artifacts_executables
```

Then, in `jobs`:
```yaml
      # ASan build and tests
      - b_ubu_asan: *workflow_trigger_on_tags
      - b_ubu_san_clang:
          <<: *workflow_trigger_on_tags
          sanitizer: address
      - t_ubu_asan_constantinople: *workflow_ubuntu2004_asan
      - t_ubu_asan_constantinople_clang: *workflow_ubuntu2004_asan_clang
      - t_ubu_asan_cli: *workflow_ubuntu2004_asan

      # UBSan build and tests
      - b_ubu_san_clang:
          <<: *workflow_trigger_on_tags
          sanitizer: undefined
      - t_ubu_ubsan_clang: *workflow_ubuntu2004_ubsan_clang
```

Or, alternatively, to avoid making `jobs` more verbose, you could still define `b_ubu_asan_clang` and `b_ubu_ubsan_clang` as standalone dicts with defaults (as they are now) but using this parameterized definition so that you only have to specify the parameter there:
```yaml
  b_ubu_asan_clang:
    <<: &b_ubu_san_clang
    sanitizer: address

  b_ubu_ubsan_clang:
    <<: &b_ubu_san_clang
    sanitizer: undefined
```

Reply:
Ah nice, thanks for the tip!